agent/utils/buffer-ring/bigqueue/bigqueue.go

```diff
@@ -179,14 +179,19 @@ func (r *Ring) Close() {
 	default:
 		close(r.done)
 		r.wg.Wait()
-		if err := r.fq.Close(); err != nil {
+		r.recvLock.Lock()
+		err := r.fq.Close()
+		r.recvLock.Unlock()
+		if err != nil {
 			r.l.Errorf("closing cache: %+v", err)
 		}
 		r.l.Info("cache closed")
 	}
 }
 
 func (r *Ring) isEmpty() bool {
+	r.recvLock.Lock()
+	r.recvLock.Unlock()
```

Comment:
the same with this method

Reply:
In the `New` func it's not covered, but that's not an issue.